src/ray/util/logging.cc

```diff
@@ -73,6 +81,59 @@ struct StdoutLogger : public google::base::Logger {
 static StdoutLogger stdout_logger_singleton;
 #endif
 
+#ifdef RAY_USE_SPDLOG
+/// NOTE(lingxuan.zlx): we reuse glog const_basename function from its utils.
+inline const char *ConstBasename(const char *filepath) {
+  const char *base = strrchr(filepath, '/');
+#ifdef OS_WINDOWS  // Look for either path separator in Windows
+  if (!base) base = strrchr(filepath, '\\');
+#endif
+  return base ? (base + 1) : filepath;
+}
+
+class SpdLogMessage final {
+ public:
+  explicit SpdLogMessage(const char *file, int line, int loglevel) : loglevel_(loglevel) {
+    stream() << ConstBasename(file) << ":" << line << ": ";
+  }
+  inline std::shared_ptr<spdlog::logger> get_logger() {
+    auto logger = spdlog::get("ray_log_sink");
+    if (!logger) {
+      logger = spdlog::get("stderr");
+    }
+    // If no default logger we just emit all log informations to stderr.
+    if (!logger) {
+      logger = spdlog::stderr_color_mt("stderr");
+      logger->set_pattern("[%Y-%m-%d %H:%M:%S,%e %L %P %t] %v");
+    }
+    return logger;
+  }
+
+  inline void Flush() {
+    auto logger = get_logger();
+    if (loglevel_ == static_cast<int>(spdlog::level::critical)) {
+      stream() << "\n" << ray::GetCallTrace();
```

Comment:
Why we need this line? I think GLOG will fetch the stack trace for us once `std::abort()` is invoked.

Reply:
> Could you paste an example output of `RAY_CHECK`?

Stacktraces will be exactly produced twice with InstallFailureSignalHandler, but nothing will be catched we do not call GetCallTrace manually without InstallFailureSignalHandler for glog.